java/src/test/java/ai/rapids/cudf/ArrowColumnVectorTest.java

```diff
@@ -0,0 +1,363 @@
+import org.apache.arrow.vector.IntVector;
+import org.apache.arrow.vector.VarCharVector;
+import org.apache.arrow.vector.complex.ListVector;
+import org.apache.arrow.vector.complex.StructVector;
+import org.apache.arrow.vector.util.Text;
+
+import org.junit.jupiter.api.Test;
+
+import static ai.rapids.cudf.TableTest.assertColumnsAreEqual;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class ArrowColumnVectorTest extends CudfTestBase {
+
+  @Test
+  void testArrowIntMultiBatches() {
+    ArrowColumnBuilder builder = new ArrowColumnBuilder(new HostColumnVector.BasicType(true, DType.INT32));
+    BufferAllocator allocator = new RootAllocator(Long.MAX_VALUE);
+    int numVecs = 4;
+    IntVector[] vectors = new IntVector[numVecs];
+    try {
+      ArrayList<Integer> expectedArr = new ArrayList<Integer>();
+      for (int j = 0; j < numVecs; j++) {
+        int pos = 0;
+        int count = 10000;
+        IntVector vector = new IntVector("intVec", allocator);
+        int start = count * j;
+        int end = count * (j + 1);
+        for (int i = start; i < end; i++) {
+          expectedArr.add(i);
+          ((IntVector) vector).setSafe(pos, i);
+          pos++;
+        }
+        vector.setValueCount(count);
+        vectors[j] = vector;
+        ByteBuffer data = vector.getDataBuffer().nioBuffer();
+        ByteBuffer valid = vector.getValidityBuffer().nioBuffer();
+        builder.addBatch(vector.getValueCount(), vector.getNullCount(), data, valid, null);
+      }
+      ColumnVector cv = builder.buildAndPutOnDevice();
+      ColumnVector expected = ColumnVector.fromBoxedInts(expectedArr.toArray(new Integer[0]));
+      assertEquals(cv.getType(), DType.INT32);
+      assertColumnsAreEqual(expected, cv, "ints");
+    } finally {
+      for (int i = 0; i < numVecs; i++) {
+        vectors[i].close();
+      }
+    }
+  }
+
+  @Test
+  void testArrowLong() {
+    ArrowColumnBuilder builder = new ArrowColumnBuilder(new HostColumnVector.BasicType(true, DType.INT64));
+    BufferAllocator allocator = new RootAllocator(Long.MAX_VALUE);
+    BigIntVector vector = new BigIntVector("vec", allocator);
+    try {
+      ArrayList<Long> expectedArr = new ArrayList<Long>();
+      int count = 10000;
+      for (int i = 0; i < count; i++) {
+        expectedArr.add(new Long(i));
+        ((BigIntVector) vector).setSafe(i, i);
+      }
+      vector.setValueCount(count);
+      ByteBuffer data = vector.getDataBuffer().nioBuffer();
+      ByteBuffer valid = vector.getValidityBuffer().nioBuffer();
+      builder.addBatch(vector.getValueCount(), vector.getNullCount(), data, valid, null);
+      ColumnVector cv = builder.buildAndPutOnDevice();
+      assertEquals(cv.getType(), DType.INT64);
+      ColumnVector expected = ColumnVector.fromBoxedLongs(expectedArr.toArray(new Long[0]));
+      assertColumnsAreEqual(expected, cv, "Longs");
+    } finally {
+      vector.close();
```

Comment:
Nit: Use try-with-resources instead of try..finally here and in the other tests.

Reply:
updated